shopinvader_schema_invoice/schemas/invoice.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Camptocamp SA
+# @author: Simone Orsi <simahawk@gmail.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from datetime import date
+from enum import Enum
+
+from extendable_pydantic import StrictExtendableBaseModel
+
+from .amount import InvoiceAmount
+
+
+class InvoiceState(Enum):
+    draft = "draft"
+    posted = "posted"
+    cancel = "cancel"
+
+
+class InvoicePaymentState(Enum):
+    not_paid = "not_paid"
+    in_payment = "in_payment"
+    paid = "paid"
+    partial = "partial"
+    reversed = "reversed"
+    invoicing_legacy = "invoicing_legacy"
+
+
+class Invoice(StrictExtendableBaseModel, extra="ignore"):
+    id: int
+    name: str
+    date_invoice: date
+    date_due: date | None = None
+    state: InvoiceState
+    payment_state: InvoicePaymentState
+    ref: str | str
+    payment_reference: str | str
+    amount: InvoiceAmount | None
+
+    @classmethod
+    def from_record(cls, odoo_rec):
```

Comment:
I would prefer to name from which odoo model we construct the fastapi model. 
```suggestion
    def from_account_move(cls, odoo_rec):
```